client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -41,18 +48,50 @@ class RequiredPluginsInstallView extends Component {
 			toInstall: [],
 			workingOn: '',
 			stepIndex: 0,
+			numTotalSteps: 0,
+			requestedTransferStatus: false,
 		};
 		this.updateTimer = false;
+		this.transferStatusFetcher = null;
 	}
 
 	componentDidMount = () => {
+		const { signupIsStore } = this.props;
+
 		this.createUpdateTimer();
+
+		if ( signupIsStore ) {
+			this.fetchAutomatedTransferStatus();
+			this.startSetup();
+		}
 	};
 
 	componentWillUnmount = () => {
 		this.destroyUpdateTimer();
 	};
 
+	fetchAutomatedTransferStatus = () => {
```

Comment:
I don't understand what this function does. Poll for the transfer status every 4 seconds until the `automatedTransferStatus` gets a non-null value? Or until the transfer is `COMPLETE`?

Is the `requestTransferStatus` variable really needed? It seems that `automatedTransferStatus` value might already contain all the information you need.

Reply:
The `requestedTransferStatus` is not needed any more, good catch. I originally needed it because I was calling the fn differently and in `componentWillReceiveProps`. However, now that we call it again only if `! automatedTransferStatus`, it's not needed.

> I don't understand what this function does. Poll for the transfer status every 4 seconds until the automatedTransferStatus gets a non-null value? Or until the transfer is COMPLETE?

So: when a customer moves through the store signup flow and purchases the Business plan, we try to initiate AT on the backend. However, we might not yet have the ssl cert for the domain so we can't initiate AT -- instead, we ask for eligibility first.

If the customer gets to the Woo installation page before AT starts, there's no AT status to get. The status endpoint will return an error, thus `automatedTransferStatus` will be `null`.

Normally, if you call `fetchAutomatedTransferStatus`, it automatically calls itself every 3 seconds until the transfer status is not `COMPLETE`. However, it won't call itself it there is no transfer status -- which is our case here.

That's why we have to call `fetchAutomatedTransferStatus` manually until we don't get any transfer status. When we get any transfer status, the function will call itself automatically until status is not `COMPLETE` so we don't need to call it manually again.